onnxruntime/core/providers/rocm/rocm_execution_provider_info.cc

```diff
@@ -25,9 +27,27 @@ const EnumNameMapping<ArenaExtendStrategy> arena_extend_strategy_mapping{
 
 ROCMExecutionProviderInfo ROCMExecutionProviderInfo::FromProviderOptions(const ProviderOptions& options) {
   ROCMExecutionProviderInfo info{};
+  void* alloc = nullptr;
+  void* free = nullptr;
 
   ORT_THROW_IF_ERROR(
       ProviderOptionsParser{}
```

Comment:
Somewhat unrelated - but should we update this to more closely match CUDA?  E.g. support for `hipGetDeviceCount()`